tests/nsrt_learning/test_side_predicate_learning.py

```diff
@@ -89,10 +97,38 @@ def test_backchaining():
                              expected_strs):
         assert str(pnad) == repr(pnad) == exp_str
 
+    # Test sidelining where an existing operator needs to
+    # be specialized.
+    goal3 = {Asleep([bob]), Sad([bob])}
+    act3 = Action([], Cry)
+    traj3 = LowLevelTrajectory([state_awake_and_sad, state_asleep_and_sad],
+                               [act3], True, 0)
+    traj4 = LowLevelTrajectory([state_awake_and_happy, state_asleep_and_sad],
+                               [act3], True, 0)
+    task3 = Task(state_awake_and_sad, goal3)
+    segment3 = Segment(traj3, set([Sad([bob])]), goal3, Cry)
+    segment4 = Segment(traj4, set(), goal3, Cry)
+    for pnad in initial_pnads:
```

Comment:
instead of this for loop, do:
```
cry_pnads = [pnad for pnad in initial_pnads if pnad.op.name == "CryOp"]
assert len(cry_pnads) == 1
cry_pnad = cry_pnads[0]
```
(it's just slightly more robust this way)